lib/HTFeed/Stage.pm

```diff
@@ -23,11 +23,14 @@ sub new {
         croak __PACKAGE__ . ' can only construct subclass objects';
     }
 
+    # No class inheriting from HTFeed::Stage needs to instantiate
+    # its own $self->{jm} for storing JobMetrics.
     my $self = {
         volume => undef,
         @_,
         has_run => 0,
         failed  => 0,
+	jm      => HTFeed::JobMetrics->new
```

Comment:
Do the tests cover the case of multiple instances of `HTFeed::JobMetrics` writing to the same file in the same process? (e.g. we have one in `QueueRunner`, then each individual `Stage` gets one..)

If this works, it might point to an approach that could work in imgsrv/mdp-lib apps as well in terms of sidestepping the need to pass around a single object or to keep track of one globally.

If this doesn't work as-is, we could perhaps change `HTFeed::JobMetrics` to implement a kind of singleton pattern where you do like `HTFeed::JobMetrics->instance` that initializes the single new one if we need it.

Reply:
Class now implements the singleton pattern, changed constructor from `new()` to `get_instance()`.

Reply:
I've added a test where multiple copies of the singleton tries to write at the same time, but having issues (test is green on github, red locally).

Reply:
Now there is a test, AND we're shamelessly green.